apimodels/cedar_models.go

```diff
@@ -1,9 +1,155 @@
 package apimodels
 
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/log"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/evergreen-ci/timber"
+	"github.com/evergreen-ci/timber/buildlogger"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/pkg/errors"
+)
+
 type CedarConfig struct {
 	BaseURL  string `json:"base_url"`
 	RPCPort  string `json:"rpc_port"`
 	Username string `json:"username"`
 	APIKey   string `json:"api_key,omitempty"`
 	Insecure bool   `json:"insecure"`
 }
+
+// GetBuildloggerLogsOptions represents the arguments passed into the
+// GetBuildloggerLogs function.
+// TODO (EVG-20019): Remove the "V2" suffix once log fetching is consolidated
+// and the original GetBuildloggerLogsOptions struct is deleted.
+type GetBuildloggerLogsOptionsV2 struct {
+	BaseURL   string   `json:"_"`
+	TaskID    string   `json:"-"`
+	Execution *int     `json:"-"`
+	TestName  string   `json:"-"`
+	Tags      []string `json:"-"`
+	Start     int64    `json:"-"`
+	End       int64    `json:"-"`
+	Limit     int      `json:"-"`
+	Tail      int      `json:"-"`
+}
+
+// GetBuildloggerLogs makes request to Cedar for a specifc log and returns a
+// log iterator.
+// TODO (EVG-20019): Remove the "V2" suffix once log fetching is consolidated
+// and the original GetBuildloggerLogs function is deleted.
+func GetBuildloggerLogsV2(ctx context.Context, opts GetBuildloggerLogsOptionsV2) (log.LogIterator, error) {
+	usr := gimlet.GetUser(ctx)
+	if usr == nil {
+		return nil, errors.New("error getting user from context")
+	}
+
+	var start, end time.Time
+	if opts.Start > 0 {
+		start = time.Unix(0, opts.Start).UTC()
+	}
+	if opts.End > 0 {
+		end = time.Unix(0, opts.Start).UTC()
+	}
+
+	getOpts := buildlogger.GetOptions{
+		Cedar: timber.GetOptions{
+			BaseURL:  fmt.Sprintf("https://%s", opts.BaseURL),
+			UserKey:  usr.GetAPIKey(),
+			UserName: usr.Username(),
+		},
+		TaskID:        opts.TaskID,
+		Execution:     opts.Execution,
+		TestName:      opts.TestName,
+		Tags:          opts.Tags,
+		PrintTime:     true,
+		PrintPriority: true,
+		Start:         start,
+		End:           end,
+		Limit:         opts.Limit,
+		Tail:          opts.Tail,
+	}
+	r, err := buildlogger.Get(ctx, getOpts)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting logs from Cedar Buildlogger")
+	}
+
+	return newBuildloggerIterator(r), nil
+}
+
+type buildloggerIterator struct {
+	readCloser io.ReadCloser
+	reader     *bufio.Reader
+	item       log.LogLine
+	catcher    grip.Catcher
+	exhausted  bool
+	closed     bool
+}
+
+func newBuildloggerIterator(r io.ReadCloser) *buildloggerIterator {
```

Comment:
Can we unit test the iterator? Especially since it seems like iteration is highly sensitive to having the proper format log lines.

Reply:
I can add tests but we don't really test this package, see my [comment above](https://github.com/evergreen-ci/evergreen/pull/7079#discussion_r1342940545).

Reply:
SGTM assuming we're deleting this code in a year.